### PR TITLE
[FMV] Add __FUNCTION_MULTI_VERSIONING_SUPPORT_LEVEL.

### PR DESCRIPTION
**Pull request by DanielKristofKiss**
Checklist: (mark with ``X`` those which apply)

* [ ] If an issue reporting the bug exists, I have mentioned it in the
      PR (do not bother creating the issue if all you want to do is
      fixing the bug yourself).
* [ ] I have added/updated the `SPDX-FileCopyrightText` lines on top
      of any file I have edited. Format is `SPDX-FileCopyrightText:
      Copyright {year} {entity or name} <{contact informations}>`
      (Please update existing copyright lines if applicable. You can
      specify year ranges with hyphen , as in `2017-2019`, and use
      commas to separate gaps, as in `2018-2020, 2022`).
* [ ] I have updated the `Copyright` section of the sources of the
      specification I have edited (this will show up in the text
      rendered in the PDF and other output format supported). The
      format is the same described in the previous item.
* [x] I have run the CI scripts (if applicable, as they might be
      tricky to set up on non-*nix machines). The sequence can be
      found in the [contribution
      guidelines](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md#continuous-integration). Don't
      worry if you cannot run these scripts on your machine, your
      patch will be automatically checked in the Actions of the pull
      request.
* [x] I have added an item that describes the changes I have
      introduced in this PR in the section **Changes for next
      release** of the section **Change Control**/**Document history**
      of the document. Create **Changes for next release** if it does
      not exist. Notice that changes that are not modifying the
      content and rendering of the specifications (both HTML and PDF)
      do not need to be listed.
* [x] When modifying content and/or its rendering, I have checked the
      correctness of the result in the PDF output (please refer to the
      instructions on [how to build the PDFs
      locally](https://github.com/ARM-software/acle/blob/main/CONTRIBUTING.md#continuous-integration)).
* [ ] The variable `draftversion` is set to `true` in the YAML header
      of the sources of the specifications I have modified.
* [ ] Please *DO NOT* add my GitHub profile to the list of contributors
      in the [README](https://github.com/ARM-software/acle/blob/main/README.md#contributors-) page of the project.

### CODE REVIEW

**Comment by DanielKristofKiss** [comment]
cc: @ilinpv @jroelofs @andrewcarlotti

**Comment by ilinpv** [comment]
Looks reasonable to me.

**Comment by jroelofs** [comment]
Ditto. Thank you!

**Comment by DanielKristofKiss** [comment]
Hi @vhscampos,
One of the reasons:
https://github.com/llvm/llvm-project/issues/79659 

If the macro reports the version then a program code could check the support level. Later if we introduce new features to the spec then we would have a way to detect it.
Also make easier to check feature level support to ( .e.g. arch FeatureX only introduced in 2025 release of ACLE )

**Comment by vhscampos** [comment]
I'm not sure this is a good idea. `__HAVE_FUNCTION_MULTI_VERSIONING` is a name that reflects a binary choice. Even having it as a bitmask would be a stretch (the previously suggested approach in https://github.com/llvm/llvm-project/issues/79659).

If we want to gate things behing conditions such as this:
`#if __HAVE_FUNCTION_MULTI_VERSIONING >= __ARM_ACLE_VERSION(2024, 1, 0 )`
Then we should just use directly:
`#if __ARM_ACLE >= __ARM_ACLE_VERSION(2024, 1, 0 )`

**Comment by jroelofs** [comment]
I am not sure that using `__ARM_ACLE >= __ARM_ACLE_VERSION(2024, 1, 0 )` directly solves the problem that I have:

> the presence of compilers in the wild that claim to but don't actually support [FMV] means we don't have a great way to gate usage of this feature via the preprocessor.

For context, in clang, support for FMV depends on:
* platform support for IFuncs, or an equivalent lowering (which isn't implemented for all platforms)
* libcompiler_rt being the runtime library

which are both orthogonal to the implemented ACLE version.

**Comment by vhscampos** [comment]
Sorry for the long delay. I had some discussions internally but the outcome got lost in the process.

I reckon that __HAVE_FUNCTION_MULTI_VERSIONING is a macro whose value should remain a boolean. Its name has the explicit meaning of a boolean piece of information.

For the purpose intended in this change, I recommend that a new macro is created with a name similar to __MULTI_VERSIONING_SUPPORT_LEVEL, with a value derived from the relevant ACLE version:

`#define __MULTI_VERSIONING_SUPPORT_EVEL __ARM_ACLE_VERSION(2024, 1, 0)`

**Comment by jroelofs** [comment]
works for me

**Comment by vhscampos** [comment]
@labrinea Can you please review this as well once you're able to?

**Comment by labrinea** [comment]
Lgtm

File: main/acle.md [hunk 1]
```diff
@@ -2535,6 +2537,15 @@ following:
 versioning mechanism described in this section is supported by the
 compiler and it is enabled.
 
+`__FUNCTION_MULTI_VERSIONING_SUPPORT_LEVEL` is defined to the currently supported
+version of the ACLE. The value and the format are the same as the `__ARM_ACLE`.
+
+For example could be implemted as:
```
Comment (vhscampos) [inline main/acle.md:2543]: Typo "implemted"

File: main/acle.md [hunk 2]
```diff
@@ -400,6 +400,8 @@ Armv8.4-A [[ARMARMv84]](#ARMARMv84). Support is added for the Dot Product intrin
 * Added a requirement for function version declaration in Function Multi Versioning.
 * Fixed some rendering issues in the online Markdown documentation and fixed
   a misplaced anchor.
+* Added __FUNCTION_MULTI_VERSIONING_SUPPORT_LEVEL to indicate the support level of the
```
Comment (vhscampos) [inline main/acle.md:403]: Please wrap the macro name between ticks for better formatting

File: main/acle.md (hunk 1 above)
Comment (vhscampos) [inline main/acle.md:2543]: Better phrasing: "For example, it can be implemented as:"

**Review by vhscampos — CHANGES_REQUESTED** [review]

File: main/acle.md (hunk 2 above)
Comment (DanielKristofKiss) [reply main/acle.md]: done

File: main/acle.md (hunk 1 above)
Comment (DanielKristofKiss) [reply main/acle.md]: done

Comment (DanielKristofKiss) [reply main/acle.md]: done

**Review by vhscampos — APPROVED** [review]
LGTM. Thank you

**Review by sallyarmneale — COMMENTED** [review]
LGTM

**Pull request merged**